google/cloud/bigtable/test_proxy/CMakeLists.txt

```diff
@@ -0,0 +1,32 @@
+# Copyright 2023 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# A minimal CMakeList.txt showing how to use the Cloud Bigtable test proxy
+# in CMake-based projects.
+
+cmake_minimum_required(VERSION 3.10...3.24)
+project(google-cloud-cpp-bigtable-test-proxy CXX)
```

Comment:
We don't want this to be its own project. The quickstarts are not the best template. Sorry for not giving you more guidance before starting this work. The `benchmarks` are probably a better template:

https://github.com/googleapis/google-cloud-cpp/blob/main/google/cloud/bigtable/benchmarks/CMakeLists.txt

You will want....

1. `add_subdirectory(test_proxy)` in the bigtable level CMakeLists.txt: https://github.com/googleapis/google-cloud-cpp/blob/71aaea39362c864b1f9f696394197ce989f62fa3/google/cloud/bigtable/CMakeLists.txt#L440

this tells cmake to recurse into the `bigtable/test_proxy` directory.

2. a `google_cloud_cpp_grpcpp_library(...)` that is the equivalent of these bazel targets: https://github.com/googleapis/google-cloud-cpp/blob/0a373eba576952b1fe970a0f723b413d5b988273/google/cloud/bigtable/test_proxy/BUILD.bazel#L26-L49

This might be an example: https://github.com/googleapis/google-cloud-cpp/blob/71aaea39362c864b1f9f696394197ce989f62fa3/generator/integration_tests/CMakeLists.txt#L27-L36

But basically you have a proto, and you need to generate the grpc library for it, so your proxy library can depend on it.

3. An `add_library(...)` which is the equivalent of this `cc_library`: https://github.com/googleapis/google-cloud-cpp/blob/0a373eba576952b1fe970a0f723b413d5b988273/google/cloud/bigtable/test_proxy/BUILD.bazel#L51-L62

e.g. https://github.com/googleapis/google-cloud-cpp/blob/bc9b10a08070f2d71be6289aafa40c0208f27211/google/cloud/bigtable/benchmarks/CMakeLists.txt#L17-L41

basically bazel `deps` are cmake `target_link_libraries`.

4. A `google_cloud_cpp_add_executable(...)` which is the equivalent of the `cc_binary`: https://github.com/googleapis/google-cloud-cpp/blob/0a373eba576952b1fe970a0f723b413d5b988273/google/cloud/bigtable/test_proxy/BUILD.bazel#L63-L69

e.g. https://github.com/googleapis/google-cloud-cpp/blob/bc9b10a08070f2d71be6289aafa40c0208f27211/google/cloud/bigtable/benchmarks/CMakeLists.txt#L96

---

There may be more, but that should get you started.

You will need to test the CMake build. To do that, I would:
1. **locally** edit this line to say `-DGOOGLE_CLOUD_CPP_ENABLE=bigtable` https://github.com/googleapis/google-cloud-cpp/blob/71aaea39362c864b1f9f696394197ce989f62fa3/ci/cloudbuild/builds/clang-tidy.sh#L42

(That way you only build the code that you care about). (Do not submit this change!)

2. Run `ci/cloudbuild/build.sh -t clang-tidy-pr`

Reply:
During my local testing, I encountered the error

```
2023-04-20T22:54:53Z (+1s): Creating a git repository
warning: re-init: ignored --initial-branch=ephemeral-branch
Reinitialized existing Git repository in /workspace/.git/
fatal: --local can only be used inside a git repository
```

I resolved it by adding `git config --global --add safe.directory /workspace` to  ci/cloudbuild/builds/lib/git.sh (it's reverted in the commit)